buildtest/buildsystem/checks.py

```diff
@@ -161,23 +161,30 @@ def is_symlink_check(builder):
     Returns:
         bool: A boolean for is_symlink status check
     """
-    assert_exists = all(os.path.islink(file) for file in builder.status["is_symlink"])
+    assert_exists = []
     console.print(
         f"[blue]{builder}[/]: Check all items:  {builder.status['is_symlink']}  for symbolic links"
     )
     for sym_link in builder.status["is_symlink"]:
+
+        expanded_sym_link = os.path.expandvars(sym_link)
+        expanded_sym_link = os.path.expanduser(expanded_sym_link)
```

Comment:
These commands are not necessary because this is already covered in resolve_path which takes into account variable and shell expansion see https://github.com/buildtesters/buildtest/blob/devel/buildtest/utils/file.py#L148-L150 

So what i would do is create another method in https://github.com/buildtesters/buildtest/blob/devel/buildtest/utils/file.py such as `is_symlink` which is similar to `is_file` but will return `True` or `False` if file is symlink.

Then what you can do is just call 

```
  if is_symlink(sym_link):
    console.print(...)
```

This way we have a portable method `is_symlink` that can be called in the codebase when needed.

Reply:
Yeah, makes sense to have a portable method. I think the above commands (expandvars, expanduser) are required because `os.path.islink` requires shell and variable expanded path to the symlink. The `resolve_path` function along with shell and variable expansion resolves the symlink as well and returns full path to the location the symlink is pointing to (`os.path.realpath` in the function resolves the symlink), which doesn't satisfy the input requirements of `os.path.islink`. Additionally, `os.path.islink` doesn't take care of broken symlinks, it returns `true` if the symlink is broken. So I have used the fully resolved symlink to check that.

Reply:
Thanks for clarification. Yeah you are right, i saw the api and we will need to keep this logic around for some time.